diff --git a/drivers/video/msm_pe/msm_fb.c b/drivers/video/msm_pe/msm_fb.c
index 7dd52f1..0b66cbf 100644
--- a/drivers/video/msm_pe/msm_fb.c
+++ b/drivers/video/msm_pe/msm_fb.c
@@ -2734,6 +2734,7 @@ static int msm_fb_pan_display(struct fb_var_screeninfo *var,
 
 		if(info->node == 0) {
 			mfd->update_fb = LAYER_FB0;
+			mfd->wait_for_vsync = LAYER_FB0;
 			mdp_set_dma_pan_info(info, dirtyPtr,
 				(var->activate == FB_ACTIVATE_VBL));
 			//Push fb0 (top)
@@ -2745,11 +2746,13 @@ static int msm_fb_pan_display(struct fb_var_screeninfo *var,
 		//Update fb1 (bottom)
 		if(info->node == 1) {
 			mfd->update_fb = LAYER_FB1;
+			mfd->wait_for_vsync = LAYER_FB1;
 			mdp_set_dma_pan_info(mfd->fbi[1], dirtyPtr,
 				(var->activate == FB_ACTIVATE_VBL));
 			mdp4_overlay_update_bottom_layer(LAYER_FB1, mfd, 0, 0 );
 		}
 		else {
+			mfd->wait_for_vsync = LAYER_FB0;
 			mfd->update_fb = LAYER_FB0;
 		}
 
